nunaliit2-js/src/main/js/nunaliit2/n2.widgetLegend.js

```diff
@@ -28,558 +28,1306 @@ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+				var y = symbol[i+1];
+
+				if( typeof minx === 'undefined' ){
+					minx = x;
+				} else if( minx > x ){
+					minx = x;
+				};
+
+				if( typeof maxx === 'undefined' ){
+					maxx = x;
+				} else if( maxx < x ){
+					maxx = x;
+				};
+
+				if( typeof miny === 'undefined' ){
+					miny = y;
+				} else if( miny > y ){
+					miny = y;
+				};
+
+				if( typeof maxy === 'undefined' ){
+					maxy = y;
+				} else if( maxy < y ){
+					maxy = y;
+				};
+			};
+
+			// Compute path, recentering the symbol and adjusting the area so
+			// it fits a bounding box of 10x10
+			var path = [],
+			transx = (minx+maxx)/2,
+			transy = (miny+maxy)/2,
+			width = maxx-minx,
+			height = maxy-miny,
+			factor = (width > height) ? width / 10 : height / 10;
+			if( factor <= 0 ){
+				factor = 1;
+			};
+			for(var i=0,e=symbol.length; i<e; i=i+2){
+				var x = symbol[i];
+				var y = symbol[i+1];
+
+				var effX = (x-transx)/factor;
+				var effY = (y-transy)/factor;
+
+				// Round to .01
+				effX = Math.floor(effX * 100) / 100;
+				effY = Math.floor(effY * 100) / 100;
+
+				if( 0 === i ){
+					path.push('M ');
+				} else {
+					path.push('L ');
+				};
+
+				path.push(''+effX);
+				path.push(' '+effY+' ');
+			};
+			path.push('Z');
+
+			return path.join('');
+		}
+	});
+
+	// LegendWidget2 The widget draws a legendWidget based on the styleRules defined inside map.json
+	// Also, this widget has a checkbox before each entry, which allows users to filter in/out the documents based on the style condition.
+	// This widget connects with a conditionalModelFilter to do the model filtering. 
+	// Two model parameters: completeChoiceParameter and selectedChoiceParameter is the messenger
+	// between this widget and the conditionalModelFilter mentioned above
+	// @Param containerId {string}
+	// @Param sourceModelId {string} sourceModelId of the conditionalModelFilter
+	// @Param labels {array} Optional parameter: if provided, only the label inside this list (must also defined valid condition inside map.json)
+	// will be rendered inside widgetLegend.
+	
+	var LegendWidget2 = $n2.Class('LegendWidget2', {
+
+			dispatchService: null,
+			
+			showService: null,
+			
+			sourceModelId: null,
+			
+			elemId: null,
+
+			selectedChoicesChangeEventName: null,
+
+			selectedChoicesSetEventName: null,
+
+			selectedChoices: null,
+			
+			selectedChoiceIdMap: null,
+
+			initialize: function(opts_){
+				var opts = $n2.extend({
+					containerId: null
+					,dispatchService: null
+					,showService: null
+					,sourceModelId: null
+					,allChoicesLabel: null
+					,tooltip: null
+					,moduleDisplay: null
+					,labels:null
+				},opts_);
+				
+				var _this = this;
+				
+				this.dispatchService = opts.dispatchService;
+				this.sourceCanvasName = opts.sourceCanvasName;
+				this.showService = opts.showService;
+				this.sourceModelId = opts.sourceModelId;
+				this.allChoicesLabel = opts.allChoicesLabel;
+				
+				this.labels = opts.labels;
```

Comment:
this.labels needs to add support for an array of localized strings. 
```javascript
[
  	{
  	  	"nunaliit_type": "localized",
  	  	"en": "Fire station",
	  	"fr": "Caserne de pompiers"
  	}
]
```

Reply:
Commit 7903140 provides support for localized object